src/sql/workbench/contrib/dashboard/browser/core/actions.ts

```diff
@@ -75,6 +75,116 @@ export class RefreshWidgetAction extends Action {
 	}
 }
 
+export class RestoreToolbarAction extends Action {
+	private static readonly ID = 'restore';
+	private static readonly LABEL = nls.localize('restore', "Restore");
+	private static readonly CSSCLASS = 'restore-toolbar dashboard-toolbar-item';
+
+	constructor(
+		private restoreFn: () => void,
+		private context: any // this
+	) {
+		super(RestoreToolbarAction.ID, RestoreToolbarAction.LABEL, RestoreToolbarAction.CSSCLASS);
+	}
+
+	run(): Promise<boolean> {
+		try {
+			this.restoreFn.apply(this.context);
+			return Promise.resolve(true);
+		} catch (e) {
+			return Promise.resolve(false);
```

Comment:
This error should really be logged somewhere, at least send something to the logger service I'd say. (same for the other actions you added)